src/promptflow/promptflow/_telemetry/activity.py

```diff
@@ -51,11 +51,16 @@ def log_activity(
     :type custom_dimensions: dict
     :return: None
     """
+    from promptflow._sdk._utils import setup_user_agent_to_operation_context
+
+    user_agent = setup_user_agent_to_operation_context()
+
     activity_info = {
         # TODO(2699383): use same request id with service caller
         "request_id": str(uuid.uuid4()),
         "activity_name": activity_name,
         "activity_type": activity_type,
+        "user_agent": user_agent,
```

Comment:
It seems that our current logs are already targeting user_agent, the user_agent added here directly overwrites the previous? Do we need to do any merging?
![image](https://github.com/microsoft/promptflow/assets/23182548/af186574-9d5e-4163-8dc4-5ae0c987a8ff)

Reply:
We'll de-dup when adding, see `append_user_agent`'s implementation. Also, we only add it when CLI/SDK starts. Most time we'll be reading it.